Java/core/src/main/java/com/amazon/randomcutforest/RandomCutForestState.java

```diff
@@ -15,55 +15,54 @@
 
 package com.amazon.randomcutforest;
 
-import java.util.ArrayList;
 import java.util.List;
 
+import lombok.Data;
+
 import com.amazon.randomcutforest.executor.Sequential;
-import com.amazon.randomcutforest.sampler.CompactSamplerData;
 import com.amazon.randomcutforest.sampler.Weighted;
-import com.amazon.randomcutforest.store.PointStoreDoubleData;
-import com.amazon.randomcutforest.tree.TreeData;
+import com.amazon.randomcutforest.state.sampler.CompactSamplerState;
+import com.amazon.randomcutforest.state.store.PointStoreDoubleState;
+import com.amazon.randomcutforest.tree.CompactRandomCutTreeState;
 
 /**
  * A class that encapsulates most of the data used in a RandomCutForest such
  * that the forest can be serialized and deserialized.
  */
-public class ForestState {
-
-    public String typeOfForest = null;
+@Data
+public class RandomCutForestState {
 
-    public long entreesSeen;
+    private long entreesSeen;
 
-    public double lambda;
+    private double lambda;
 
-    public int numberOfTrees;
+    private int numberOfTrees;
 
-    public int sampleSize;
+    private int sampleSize;
 
-    public int dimensions;
+    private int dimensions;
 
-    public int outputAfter;
+    private int outputAfter;
 
-    public boolean storeSequenceIndexesEnabled;
+    private boolean storeSequenceIndexesEnabled;
 
-    public boolean compactEnabled;
+    private boolean compactEnabled;
 
-    public boolean centerOfMassEnabled;
+    private boolean centerOfMassEnabled;
 
-    public boolean parallelExecutionEnabled;
+    private boolean parallelExecutionEnabled;
 
-    public int threadPoolSize;
+    private int threadPoolSize;
 
     public boolean saveTreeData;
 
-    public PointStoreDoubleData pointStoreDoubleData;
-
-    public ArrayList<CompactSamplerData> compactSamplerData;
+    public PointStoreDoubleState pointStoreDoubleState;
 
-    public ArrayList<List<Sequential<double[]>>> sequentialSamplerData;
+    public List<CompactSamplerState> compactSamplerStates;
```

Comment:
I'd like to leave this for now. The previous PR added methods for creating state classes and this one moves and rewrites some of that logic into the new mapper classes, but I didn't remove the old  code and I didn't complete the rewrite because the mapper for the RandomCutForest class doesn't implement the toModel() method yet. If I make this change, it will require additional changes in code that will be deleted in a future PR.